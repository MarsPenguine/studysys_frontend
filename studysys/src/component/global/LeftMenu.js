import React from "react";
import { Menu } from "antd";
import Link from "next/link";
const ROUTE_PATH = "home";
const { SubMenu } = Menu;

const renderMenu = (menu) => {
  return (
    <SubMenu
      key={menu.key}
      title={
        <span>
          <menu.icon />
          <span>{menu.title}</span>
        </span>
      }
    >
      {renderMenuItems(menu.menuItems)}
    </SubMenu>
  );
};

function renderMenuItems(menuItems) {
  return menuItems.map((menuItem) => (
    <Menu.Item key={menuItem.key}>
      <Link
        href={`/${ROUTE_PATH}/${menuItem.category}/${menuItem.subcategory}`}
      >
        <a>
          <menuItem.icon />
          <span>{menuItem.title}</span>
        </a>
      </Link>
    </Menu.Item>
  ));
}

export default function LeftMenu(props) {
  const { location, locationString, homeLeftBarConfig } = props;

  return (
    <Menu
      theme="dark"
      defaultSelectedKeys={[locationString]}
      mode="inline"
      defaultOpenKeys={[location[2]]}
    >
      {renderMenuItems(homeLeftBarConfig.getHome())}
      {renderMenu(homeLeftBarConfig.getStudent())}
      {renderMenu(homeLeftBarConfig.getCourse())}
    </Menu>
  );
}
