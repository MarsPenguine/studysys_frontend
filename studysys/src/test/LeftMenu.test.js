import React from "react";

import {shallow} from "enzyme";
import LeftMenuTestConfig from "./LeftMenuTestConfig";
import LeftMenu from "../component/global/LeftMenu";

/**
 * <LeftMenu location={location} locationString={locationString} />
 * @type {string}
 */

//test suite
describe("Test Left Menu Function", () => {
  //test case
  it("Left Menu render", () => {
    const props = {
      location: ["course", "list"],
      locationString: "course list",
      homeLeftBarConfig: LeftMenuTestConfig,
    };

    const component = shallow(<LeftMenu {...props} />);
    expect(component).toMatchSnapshot();
  });
});
