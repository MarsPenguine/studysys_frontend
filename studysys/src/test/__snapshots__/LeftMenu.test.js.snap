// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Left Menu Function Left Menu render 1`] = `
<Menu
  defaultOpenKeys={
    Array [
      undefined,
    ]
  }
  defaultSelectedKeys={
    Array [
      "course list",
    ]
  }
  mode="inline"
  theme="dark"
>
  <MenuItem
    key="0"
  >
    <Link
      href="/home/undefined/undefined"
    >
      <a>
        <icon />
        <span>
          Home
        </span>
      </a>
    </Link>
  </MenuItem>
  <SubMenu
    key="student"
    title={
      <span>
        <icon />
        <span>
          Students
        </span>
      </span>
    }
  >
    <MenuItem
      key="student list"
    >
      <Link
        href="/home/student/list"
      >
        <a>
          <icon />
          <span>
            Student List
          </span>
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="student add"
    >
      <Link
        href="/home/student/add"
      >
        <a>
          <icon />
          <span>
            Add Student
          </span>
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="student selection"
    >
      <Link
        href="/home/student/selection"
      >
        <a>
          <icon />
          <span>
            Selection
          </span>
        </a>
      </Link>
    </MenuItem>
  </SubMenu>
  <SubMenu
    key="course"
    title={
      <span>
        <icon />
        <span>
          Course
        </span>
      </span>
    }
  >
    <MenuItem
      key="course list"
    >
      <Link
        href="/home/course/list"
      >
        <a>
          <icon />
          <span>
            Course List
          </span>
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="course add"
    >
      <Link
        href="/home/course/add"
      >
        <a>
          <icon />
          <span>
            Add Course
          </span>
        </a>
      </Link>
    </MenuItem>
    <MenuItem
      key="course type"
    >
      <Link
        href="/home/course/type"
      >
        <a>
          <icon />
          <span>
            Course Type
          </span>
        </a>
      </Link>
    </MenuItem>
  </SubMenu>
</Menu>
`;
